workflows/tf-object-detection-training/utils.py

```diff
@@ -0,0 +1,158 @@
+        pipeline_config.model.faster_rcnn.first_stage_nms_iou_threshold = float(model_params['first_stage_nms_iou_threshold'])
+        pipeline_config.model.faster_rcnn.first_stage_max_proposals = int(model_params['first_stage_max_proposals'])
+        pipeline_config.model.faster_rcnn.first_stage_localization_loss_weight = float(model_params['first_stage_localization_loss_weight'])
+        pipeline_config.model.faster_rcnn.first_stage_objectness_loss_weight = float(model_params['first_stage_objectness_loss_weight'])
+        pipeline_config.model.faster_rcnn.initial_crop_size = int(model_params['initial_crop_size'])
+        pipeline_config.model.faster_rcnn.maxpool_kernel_size = int(model_params['maxpool_kernel_size'])
+        pipeline_config.model.faster_rcnn.maxpool_stride = int(model_params['maxpool_stride'])        
+        pipeline_config.model.faster_rcnn.second_stage_box_predictor.mask_rcnn_box_predictor.fc_hyperparams.regularizer.l2_regularizer.weight = float(model_params['second_stage_regularizer_weight'])
+        pipeline_config.model.faster_rcnn.second_stage_box_predictor.mask_rcnn_box_predictor.fc_hyperparams.initializer.variance_scaling_initializer.factor = float(model_params['second_stage_initializer_factor'])
+        pipeline_config.model.faster_rcnn.second_stage_box_predictor.mask_rcnn_box_predictor.fc_hyperparams.initializer.variance_scaling_initializer.mode = int(model_params['second_stage_initializer_mode'])
+        pipeline_config.model.faster_rcnn.second_stage_box_predictor.mask_rcnn_box_predictor.use_dropout = model_params['second_stage_use_dropout']
+        pipeline_config.model.faster_rcnn.second_stage_box_predictor.mask_rcnn_box_predictor.dropout_keep_probability = float(model_params['second_stage_dropout_keep_probability'])
+        pipeline_config.model.faster_rcnn.second_stage_post_processing.batch_non_max_suppression.score_threshold = float(model_params['second_stage_nms_score_threshold'])
+        pipeline_config.model.faster_rcnn.second_stage_post_processing.batch_non_max_suppression.iou_threshold = float(model_params['second_stage_nms_iou_threshold'])
+        pipeline_config.model.faster_rcnn.second_stage_post_processing.batch_non_max_suppression.max_detections_per_class = int(model_params['second_stage_max_detections_per_class'])
+        pipeline_config.model.faster_rcnn.second_stage_post_processing.batch_non_max_suppression.max_total_detections = int(model_params['second_stage_max_detections_max_total_detections'])
+        pipeline_config.model.faster_rcnn.second_stage_localization_loss_weight = float(model_params['second_stage_localization_loss_weight'])
+        pipeline_config.model.faster_rcnn.second_stage_classification_loss_weight = float(model_params['second_stage_classification_loss_weight'])
+
+        pipeline_config.train_config.optimizer.momentum_optimizer.learning_rate.manual_step_learning_rate.initial_learning_rate = float(model_params['initial-learning-rate'])
```

Comment:
Can we use `_` for these as well but also provide backwards compatibility for previous Workflows?

Reply:
Done! Added legacy compatibility too. Supported "initial-learning-rate" and "learning_rate" too as was in docs.